src/model/core/EthqlBlock.ts

```diff
@@ -65,8 +66,10 @@ class EthqlBlock implements EthqlBlock {
     return this._transactions.length;
   }
 
-  public transactionAt(args): EthqlTransaction {
-    return this._transactions[args.index];
+  public async transactionAt(args, { web3 }: EthqlContext): Promise<EthqlTransaction> {
+    if (args.index < 0) { return null; }
```

Comment:
Good catch adding a guard here. Maybe we should add a test for this case?

Reply:
SGTM